src/components/InterfaceEditor.tsx

```diff
@@ -270,14 +270,61 @@ const checkInterfaceHasMajorChanges = (
     const draftMapping = draftInterface.mappings.find(
       ({ endpoint }) => endpoint === initialMapping.endpoint,
     );
-    return _.isEqual(draftMapping, initialMapping);
+
+    const sensitiveMappingProperties: Array<keyof AstarteMapping> = [
+      'endpoint',
+      'type',
+      'allowUnset',
+      'reliability',
+      'databaseRetentionPolicy',
+      'databaseRetentionTtl',
+    ];
+    if (!draftMapping) {
+      return false;
+    }
+    const hasExistingMappingsChange = sensitiveMappingProperties.some(
+      (property) => draftMapping[property] !== initialMapping[property],
+    );
+    if (hasExistingMappingsChange) {
+      return false;
+    }
+    return true;
   });
   if (!hasExistingMappings) {
     return true;
   }
   return false;
 };
 
+const detectInsensitiveMappingPropertiesChange = (
+  initialInterface: AstarteInterface,
+  draftInterface: AstarteInterface,
+) => {
+  const hasMappingsChange = initialInterface.mappings.every((initialMapping) => {
+    const draftMapping = draftInterface.mappings.find(
+      ({ endpoint }) => endpoint === initialMapping.endpoint,
+    );
+    const insensitiveMappingProperies: Array<keyof AstarteMapping> = [
+      'description',
+      'documentation',
+      'explicitTimestamp',
+      'retention',
+      'expiry',
+    ];
+    if (!draftMapping) {
+      return false;
+    }
+    const hasInsensitiveMappingChange = insensitiveMappingProperies.some(
+      (property) => draftMapping[property] !== initialMapping[property],
+    );
+    if (hasInsensitiveMappingChange && draftInterface.minor <= initialInterface.minor) {
+      return false;
+    }
+    return true;
```

Comment:
The code is returning inverted boolean values.

If no `draftMapping` could be found, then there are changes (hence also minor changes).

```typescript
if (!draftMapping) {
  return true;
}
```

The second conditional can be replaced with just:
```typescript
const hasMinorMappingChanges = insensitiveMappingProperties.some(
  (property) => draftMapping[property] !== initialMapping[property],
);
return hasMinorMappingChanges;
```